lib/mobility-core/src/Kernel/Utils/Shutdown.hs

```diff
@@ -57,7 +59,7 @@ untilShutdown =
 isRunning ::
   ( MonadIO m,
     MonadReader r m,
-    HasField "isShuttingDown" r Shutdown
+    HasField' "isShuttingDown" r Shutdown
   ) =>
   m Bool
-isRunning = liftIO . atomically . isEmptyTMVar =<< asks (.isShuttingDown)
+isRunning = liftIO . atomically . isEmptyTMVar =<< asks (^. field' @"isShuttingDown")
```

Comment:
can we use the same syntax everywhere asks (^. #isShuttingDown)
maybe it will be possible with another constraint?
`type HasFieldSuper name r ty = (GHC.HasField name r ty, GL.Field name r r ty ty)`